tests/rule_based_profiler/parameter_builder/test_numeric_metric_range_multi_batch_parameter_builder.py

```diff
@@ -326,13 +334,224 @@ def test_oneshot_numeric_metric_range_multi_batch_parameter_builder_bobby(
+    )
+    p_value: float = ks_result[1]
+    assert p_value > 9.5e-1
+
+
+def test_oneshot_numeric_metric_range_multi_batch_parameter_builder_with_evaluation_dependency_bobby(
+    bobby_columnar_table_multi_batch_deterministic_data_context,
+):
+    data_context: DataContext = (
+        bobby_columnar_table_multi_batch_deterministic_data_context
+    )
+
+    batch_request: dict = {
+        "datasource_name": "taxi_pandas",
+        "data_connector_name": "monthly",
+        "data_asset_name": "my_reports",
+    }
+
+    metric_domain_kwargs: dict = {"column": "fare_amount"}
+
+    fully_qualified_parameter_name_for_value: str = "$parameter.column_min_range"
+
+    expected_value_dict: dict
+    actual_value_dict: dict
+
+    my_column_min_metric_multi_batch_parameter_builder_config: ParameterBuilderConfig = ParameterBuilderConfig(
+        module_name="great_expectations.rule_based_profiler.parameter_builder",
+        class_name="MetricMultiBatchParameterBuilder",
+        name="my_column_min",
+        metric_name="column.min",
+        metric_domain_kwargs=DOMAIN_KWARGS_PARAMETER_FULLY_QUALIFIED_NAME,
+        metric_value_kwargs=None,
+        enforce_numeric_metric=True,
+        replace_nan_with_zero=True,
+        reduce_scalar_metric=True,
+        evaluation_parameter_builder_configs=None,
+    )
+    evaluation_parameter_builder_configs: Optional[List[ParameterBuilderConfig]] = [
+        my_column_min_metric_multi_batch_parameter_builder_config,
+    ]
+    numeric_metric_range_parameter_builder: ParameterBuilder = (
+        NumericMetricRangeMultiBatchParameterBuilder(
+            name="column_min_range",
+            metric_name=None,
+            metric_multi_batch_parameter_builder_name="my_column_min",
+            metric_domain_kwargs=DOMAIN_KWARGS_PARAMETER_FULLY_QUALIFIED_NAME,
+            estimator="oneshot",
+            include_estimator_samples_histogram_in_details=True,
+            false_positive_rate=1.0e-2,
+            round_decimals=1,
+            evaluation_parameter_builder_configs=evaluation_parameter_builder_configs,
+            data_context=data_context,
+        )
+    )
+
+    variables: Optional[ParameterContainer] = None
+
+    domain: Domain = Domain(
+        domain_type=MetricDomainTypes.COLUMN,
+        domain_kwargs=metric_domain_kwargs,
+        rule_name="my_rule",
+    )
+    parameter_container: ParameterContainer = ParameterContainer(parameter_nodes=None)
+    parameters: Dict[str, ParameterContainer] = {
+        domain.id: parameter_container,
+    }
+
+    assert parameter_container.parameter_nodes is None
+
+    numeric_metric_range_parameter_builder.build_parameters(
+        domain=domain,
+        variables=variables,
+        parameters=parameters,
+        batch_request=batch_request,
+    )
+
+    parameter_nodes: Optional[Dict[str, ParameterNode]] = (
+        parameter_container.parameter_nodes or {}
+    )
+    assert len(parameter_nodes) == 1
+
+    expected_value_dict = {
```

Comment:
`Dict[str, Union[None, dict]` to be more precise?

Reply:
@NathanFarmer I do not mind applying them uniformly in another PR -- I followed the existing style.  Also, note the important bug fixes in that test module.

Reply:
@NathanFarmer I cleaned them all up -- in next commit.